### PR TITLE
Create roleinfo command

### PR DESCRIPTION
**Pull request by FlitzDev**

### CODE REVIEW

**Comment by FlitzDev** [comment]
Should mention that this is what it looks like:
image
image
image

File: src/commands/utility/RoleInfoCommand.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          if (role.permissions.has('ADMINISTRATOR')) {
+              permissions = `Administrator`
+          } if (!role.permissions.has('ADMINISTRATOR')) {
+              permissions = role.permissions.toArray().toString()
+              permissions = permissions.toLowerCase()
+              permissions = permissions.replace(/[-_]/g, ' ')
+              permissions = permissions.replace(/[,]/g, ", ")
+          } if (!permissions) {
+              permissions = `None`
+
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:39]: this is indented one step too far

File: src/commands/utility/RoleInfoCommand.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      }
+
+          const e = new Discord.MessageEmbed()
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:42]:
```suggestion
          const embed = new Discord.MessageEmbed()
```

File: src/commands/utility/RoleInfoCommand.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+              .setTitle(`About role ${role.name}`)
+              .setColor(role.color)
+              .setDescription(`
+${generic}
+
+**Permissions:** ${permissions}
+              `)
+
+          await message.channel.send(e)
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:51]:
```suggestion
          await message.channel.send(embed);
```

File: src/commands/utility/RoleInfoCommand.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+              permissions = permissions.replace(/[,]/g, ", ")
+          } if (!permissions) {
+              permissions = `None`
+
+      }
+
+          const e = new Discord.MessageEmbed()
+              .setTitle(`About role ${role.name}`)
+              .setColor(role.color)
+              .setDescription(`
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:45]: you shouldn't stretch a string over multiple lines. Use \n and + to concat multiple strings

**Review by JulianVennen — CHANGES_REQUESTED** [review]

File: src/commands/utility/RoleInfoCommand.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      }
+
+      let role = this.message.guild.roles.resolve(this.args[0]);
+      if (!role) return await this.message.channel.send(`This is not a valid role ID.`)
+
+
+      let permissions;
+      if (role.permissions.has('ADMINISTRATOR')) {
+          permissions = `Administrator`
+      } if (!role.permissions.has('ADMINISTRATOR')) {
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:24]:
```suggestion
      } 
      if (!role.permissions.has('ADMINISTRATOR')) {
```

File: src/commands/utility/RoleInfoCommand.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          permissions = role.permissions.toArray().toString()
+          permissions = permissions.toLowerCase()
+          permissions = permissions.replace(/[-_]/g, ' ')
+          permissions = permissions.replace(/[,]/g, ", ")
+      } if (!permissions) {
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:30]:
```suggestion
      } 
      if (!permissions) {
```

File: src/commands/utility/RoleInfoCommand.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                              `**Managed:** ${role.managed ? 'Yes' : 'No'}\n` +
+                              `**Position:** ${role.position} (from below)\n` +
+                              `**Hoisted:** ${role.hoist ? 'Yes' : 'No'}\n` +
+                              `**Color:** \`${role.hexColor}\` (\`${role.color}\`)\n` +
+                              `**Permissions:** ${permissions}`)
+
+          await this.message.channel.send(embed);
+
+
+      }
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:49]:
```suggestion
          await this.message.channel.send(embed);
      }
```

File: src/commands/utility/RoleInfoCommand.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  static usage = '<RoleID>';
+  
+  static description = 'Get information about a role';
+  
+  async execute() {
+      if (!this.args.length) {
+          return await this.message.channel.send(`You have to provide a role ID ${message.author}!`)
+      }
+
+      let role = this.message.guild.roles.resolve(this.args[0]);
```
Comment (JulianVennen) [inline src/commands/utility/RoleInfoCommand.js:17]: you could also accept role mentions by using util.roleMentionToId

**Review by JulianVennen — CHANGES_REQUESTED** [review]

**Review by JulianVennen — APPROVED** [review]

**Pull request merged**